src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4499,42 +4650,45 @@ internal class LocalState : AbstractLocalState
         internal struct LocalState : AbstractLocalState
 #endif
         {
-            // Consider storing nullability rather than non-nullability
-            // or perhaps expose as nullability from `this[int]` even if stored as non-nullability.
-            private BitVector _knownNullState; // No diagnostics should be derived from a variable with a bit set to 0.
-            private BitVector _notNull;
+            private ArrayBuilder<NullableAnnotation> _state;
+            public bool Reachable { get; set; }
 
-            internal LocalState(BitVector knownNullState, BitVector notNull)
+            internal LocalState(bool reachable, ArrayBuilder<NullableAnnotation> state)
             {
-                Debug.Assert(!knownNullState.IsNull);
-                Debug.Assert(!notNull.IsNull);
-                this._knownNullState = knownNullState;
-                this._notNull = notNull;
+                this.Reachable = reachable;
+                this._state = state;
             }
 
-            internal int Capacity => _knownNullState.Capacity;
+            internal int Capacity => _state?.Count ?? 0;
 
             internal void EnsureCapacity(int capacity)
             {
-                _knownNullState.EnsureCapacity(capacity);
-                _notNull.EnsureCapacity(capacity);
+                if (_state == null)
+                {
+                    _state = new ArrayBuilder<NullableAnnotation>(capacity);
+                }
+
+                if (_state.Count < capacity)
```

Comment:
Why not use `EnsureCapacity`?

Reply:
> Why not use EnsureCapacity?

It doesn't look like that would actually increase the number of elements in the builder.

---
In reply to: [231736974](https://github.com/dotnet/roslyn/pull/30913#discussion_r231736974) [](ancestors = 231736974)